addons/stock/models/stock_warehouse.py

```diff
@@ -869,6 +869,7 @@ def _check_reception_resupply(self, new_location):
 
     def _update_name_and_code(self, new_name=False, new_code=False):
         if new_code:
+            super(Warehouse, self).write({'code': new_code})
```

Comment:
Same for `new_name` since it's also needed in `_get_sequence_values`.
But I would pass the two values to `_get_sequence_values` instead of doing `write` here.